content/advanced_usage/pipelines_as_code.md

```diff
@@ -70,3 +70,87 @@ Once you've added a config repository, you'll see new pipelines in the pipeline
+
+Previously, giving access to a config repository’s backing SCM repository meant giving full access to the GoCD instance, almost. Such users have access to all environments/pipeline group and can, possibly, add tasks to expose the secrets configured.
+
+With `rules`, the owners of the config repositories can define which all environment/pipeline group can the repo add a pipeline to. Similarly, they can also define which pipelines can be referred as an upstream dependency. Hence, providing more control over what all entities can come via the config repository.
+
+- A restrictive model is followed while parsing a config repository. In the absence of a rule, no entity is accessible by default.
+
+    > The config repository defined in the example cannot refer any entity in GoCD.
+
+    ```xml
+    <config-repo id='teamA_repo' pluginId='yaml.config.plugin'>
+       <configuration>
+          ...
+      </configuration>
+      <rules/>
+    </secretConfig>
+    ```
+
+- Wildcards **(*)** in **type**:
+
+    Using a wildcard **(*)** for type implies a given rule applies to all entity types. In this case, the supported entities are `environment`, `pipeline_group` and `pipeline`.
+
+    > In the given example, a config repository can refer to any `pipeline_group` or `environment` or `pipeline` with the name `production`.
+
+    ```xml
+    <rules>
+      <allow type="*" action="refer">production</allow>
+    </rules>
+    ```
+
+    **Note**: `type` can have a wildcard(`*`) but it will not support pattern matching e.g. `pipe*`.
+
+- Wildcards **(*)** in **action**:
+
+    Using a wildcard **(*)** for action implies a given rule applies to any action on the config repository. Currently `refer` is the only supported action.
+
+    **Note**: `action` can have a wildcard(`*`) but it will not support pattern matching e.g. `ref*`.
+
+- Wildcards **(*)** in **resource**:
+
+    Resource name supports the wildcard characters '?' and '*' to represent a single or multiple (zero or more) wildcard characters.
+
+    | Wildcard Matcher   | Resource names                                                                         |
+    |------------------- | -------------------------------------------------------------------------------------- |
+    | `*_group`          | Matches `my_group` and `someother_group`, but not `testgroup` or `group1`.             |
+    | `Production_*`     | Matches `Production_Team_A` and `Production_Team_B` but not `Team_ABC_Production_D`.   |
+    | `*group*`          | Matches `group`, `my_group` and `group_A`, but not `groABCup`.                         |
+    | `Team_?_group`     | Matches `Team_A_group`, `Team_B_group` but not `Team_ABC_group` or `Team__group`.      |
+
+- `deny` is given more precedence than `allow`. Hence, if any entity is present with an explicit `deny`, that entity will be rejected.
```

Comment:
Is that true, or is it just about order as you show below? I mean, in your second example below, could I say that `allow` has more precedence, since it "overrode" a later `deny`? Or, is it just because of the order?

Reply:
Sorry for the confusion. This is just bcoz of the order in which the rules were defined. Will remove this point.